tests/extension/oneapi_device_global/device_global_queue_copy.cpp

```diff
@@ -0,0 +1,254 @@
+/*******************************************************************************
+//
+//  SYCL 2020 Conformance Test Suite
+//
+//  Provides functional test for overloads for queue::copy for device_global
+//
+*******************************************************************************/
+
+#include "../../common/common.h"
+#include "../../common/type_coverage.h"
+#include "../../usm/usm_api.h"
+#include "device_global_common.h"
+#include "type_pack.h"
+
+#define TEST_NAME device_global_queue_copy
+
+namespace TEST_NAMESPACE {
+using namespace sycl_cts;
+using namespace device_global_common_functions;
+
+#if defined(SYCL_EXT_ONEAPI_PROPERTY_LIST) && \
+    defined(SYCL_EXT_ONEAPI_DEVICE_GLOBAL)
+namespace oneapi = sycl::ext::oneapi;
+
+template <typename T>
+struct check_copy_to_dg_kernel;
+
+template <typename T>
+struct change_dg_kernel;
+
+// Creating instance with default constructor
+template <typename T>
+oneapi::device_global<T> dev_global1;
+
+template <typename T>
+oneapi::device_global<T> dev_global2;
+
+template <typename T>
+oneapi::device_global<T> dev_global3;
+
+/** @brief The function tests that queue copy overloads correctly copy to
+ * device_global
+ *  @tparam T Type of underlying struct
+ */
+template <typename T>
+void run_test_copy_to_device_global(util::logger& log,
+                                    const std::string& type_name) {
+  using element_type = std::remove_all_extents_t<T>;
+  T data{};
+  value_helper<T>::change_val(data, 1);
+  const element_type* src_data = pointer_helper(data);
```

Comment:
`const auto`?

Reply:
Done here and changed to `auto` in copy_from_device_global check.